postgres/README.md

```diff
@@ -7,56 +7,130 @@
 
-### Configuration
+1. [Configure database parameters](#database-configuration)
+1. [Grant the Datadog Agent access to the database](#agent-database-access)
+1. [Install the Datadog Agent](#agent-installation)
+1. [Configure the Datadog Agent](#agent-configuration)
 
-#### Prepare Postgres
+### Database Configuration
 
-To get started with the PostgreSQL integration, create a read-only `datadog` user with proper access to your PostgreSQL server. Start `psql` on your PostgreSQL database.
+Configure the following [Postgres parameters][35]. These parameters are required only for [Deep Database Monitoring](#deep-database-monitoring). **Note:** A server restart is required for these settings to take effect. For further information on these parameters, please see the [documentation][34].
 
-For PostgreSQL version 10 and above, run:
+| Parameter | Value | Description |
+| --- | --- | --- |
+| `shared_preload_libraries` | `pg_stat_statements` | Required for `postgresql.queries.*` metrics. Enables collection of query metrics via the the [pg_stat_statements][34] extension. |
+| `track_activity_query_size` | `4096` | Required for collection of larger queries. Increases the size of SQL text in `pg_stat_activity` and `pg_stat_statements`. If left at the default value then queries longer than `1024` characters will not be collected. |
+| `pg_stat_statements.track` | `ALL` | Optional. Enables tracking of statements within stored procedures and functions. |
+| `pg_stat_statements.max` | `10000` | Optional. Increases the number of normalized queries tracked in `pg_stat_statements`. This setting is recommended for high-volume databases which see many different types of queries from many different clients. |
 
-```shell
-create user datadog with password '<PASSWORD>';
-grant pg_monitor to datadog;
-grant SELECT ON pg_stat_database to datadog;
+##### How to configure
+
+| Deployment Method | Configuration |
+| --- | --- |
+| Self-hosted | [Parameters][36] are configured in the `postgresql.conf` file. |
+| Amazon [RDS][37] or [RDS Aurora][38] | [DB Parameter Group][39] <br/>  Note that for RDS Aurora `pg_stat_statements` is enabled by default. |
+| Google [CloudSQL][40] | [Database Flags][41] |
+
+
+### Agent Database Access
+
+The Datadog Agent requires read-only access to the database in order to collect statistics and queries.
+Connect to the **postgres** database and run the following SQL commands with superuser or a user with sufficient permissions to execute these commands. This can be done manually with [psql][43] or using any automated configuration you use for managing your databases.
+
+<!-- xxx tabs xxx -->
+<!-- xxx tab "Postgres ≥ 10" xxx -->
+
+```SQL
+CREATE USER datadog WITH password '<PASSWORD>';
+CREATE SCHEMA datadog;
+GRANT USAGE ON SCHEMA datadog TO datadog;
+GRANT USAGE ON SCHEMA public TO datadog;
+GRANT pg_monitor TO datadog;
+CREATE EXTENSION IF NOT EXISTS pg_stat_statements;
 ```
 
```

Comment:
```suggestion

Additionally, if you configure the agent to connect to a database other than the default database `postgres`, enable `pg_stat_statements` in that database as well:

    ```bash
    psql -h mydb.example.com -d customdb -U postgres -c "CREATE EXTENSION IF NOT EXISTS pg_stat_statements;"
    ```
```

Reply:
I don't think this very useful because it's not enough to only do `pg_stat_statements`. They have to create the stored procedures as well. And it's probably not worth repeating all of the setup again here.